service/deployer/installer/configurer/configmap/configmap.go

```diff
@@ -0,0 +1,144 @@
+package configmap
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	microerror "github.com/giantswarm/microkit/error"
+	micrologger "github.com/giantswarm/microkit/logger"
+	"github.com/giantswarm/operatorkit/client/k8s"
+
+	"github.com/giantswarm/draughtsman/service/deployer/installer/configurer/spec"
+)
+
+// ConfigmapConfigurerType is a Configurer that is backed by a Kubernetes
+// Configmap.
+var ConfigmapConfigurerType spec.ConfigurerType = "ConfigmapConfigurer"
+
+// Config represents the configuration used to create a Configmap Configurer.
+type Config struct {
+	// Dependencies.
+	Logger micrologger.Logger
+
+	// Settings.
+	Address     string
+	CAFilePath  string
+	CrtFilePath string
+	InCluster   bool
+	Key         string
+	KeyFilePath string
+	Name        string
+	Namespace   string
+}
+
+// DefaultConfig provides a default configuration to create a new Configmap
+// Configurer by best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		Logger: nil,
+	}
+}
+
+// New creates a new configured Configmap Configurer.
+func New(config Config) (*ConfigmapConfigurer, error) {
+	if !config.InCluster {
+		if config.Address == "" {
+			return nil, microerror.MaskAnyf(invalidConfigError, "address must not be empty")
+		}
+	}
+
+	if config.Key == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "key must not be empty")
+	}
+	if config.Name == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "name must not be empty")
+	}
+	if config.Namespace == "" {
+		return nil, microerror.MaskAnyf(invalidConfigError, "namespace must not be empty")
+	}
+
+	k8sConfig := k8s.Config{
+		Logger: config.Logger,
+
+		Address:   config.Address,
+		InCluster: config.InCluster,
+		TLS: k8s.TLSClientConfig{
+			CAFile:  config.CAFilePath,
+			CrtFile: config.CrtFilePath,
+			KeyFile: config.KeyFilePath,
+		},
+	}
+	client, err := k8s.NewClient(k8sConfig)
```

Comment:
IMO the k8s client should be injected as dependency via its configuration. This would make the interface more nice and easy. In case another service package in draughtsman will also want to use a k8s client, we would now need to inject the same settings somewhere else to create another client in a different place again. Lets create a k8s client ones and inject it where needed.